source/containers.tex

```diff
@@ -15032,3 +15034,2422 @@
 Equivalent to: \tcode{return R\{reinterpret_cast<byte*>(s.data()), s.size_bytes()\};}
 where \tcode{R} is the return type.
 \end{itemdescr}
+
+\rSec2[mdspan.syn]{Header \tcode{<mdspan>} synopsis}
+
+\indexheader{mdspan}%
+\begin{codeblock}
+namespace std {
+  // \ref{mdspan.extents}, class template \tcode{extents}
+  template<class SizeType, size_t... Extents>
+    class extents;
+
+  template<class SizeType, size_t Rank>
+    using dextents = @\seebelow@;
+
+  // \ref{mdspan.layout}, layout mapping policies
+  struct layout_left;
+  struct layout_right;
+  struct layout_stride;
+
+  // \ref{mdspan.accessor.default}, class template \tcode{default_accessor}
+  template<class ElementType>
+    class default_accessor;
+
+  // \ref{mdspan.mdspan}, class template \tcode{mdspan}
+  template<class ElementType, class Extents, class LayoutPolicy = layout_right,
+           class AccessorPolicy = default_accessor<ElementType>>
+    class mdspan;
+}
+\end{codeblock}
+
+\rSec2[mdspan.overview]{Overview}
+
+\pnum
+A \defnadj{multidimensional}{index space} is
+a Cartesian product of integer intervals.
+Each interval can be represented by a half-open range $[L_i, U_i)$,
```

Comment:
`\range` is for iterator ranges, and puts its start and end into `\tcode`. The ranges discussed here are not iterator ranges.

Reply:
There's examples of non-iterator ranges and the use of math fonts for its arguments.

```LaTeX
algorithms.tex:where \tcode{n} is the smallest integer in \range{0}{$N$} such that $E$ holds,
algorithms.tex:$E(i)$, where $i$ is the smallest integer in \range{0}{$N$}
algorithms.tex:For each integer \tcode{K} in \range{0}{last - first}
algorithms.tex:For each integer \tcode{K} in \range{0}{last - first}
algorithms.tex:For each integer \tcode{K} in \range{0}{last - first}
algorithms.tex:For each integer \tcode{K} in \range{0}{last - first}
iterators.tex:If \tcode{n > 0}, \range{i}{bound} denotes a range.
iterators.tex:If \tcode{n == 0}, \range{i}{bound} or \range{bound}{i} denotes a range.
iterators.tex:If \tcode{n < 0}, \range{bound}{i} denotes a range,
threads.tex:For all \tcode{i} in \range{0}{sizeof...(MutexTypes)},
utilities.tex:for all \tcode{i} in the range \range{0}{N}.
utilities.tex:\range{0}{sizeof...(Types)} in order, $\tcode{T}_i$
utilities.tex:In the function descriptions that follow, let $i$ be in the range \range{0}{sizeof...\brk{}(Types)}
utilities.tex:In the descriptions that follow, let $i$ be in the range \range{0}{sizeof...(Types)},

expressions.tex:\crange{-32768}{+32767}, the implementation cannot rewrite this
strings.tex:\tcode{i} in \crange{0}{size()}.
strings.tex:\tcode{i} in \crange{0}{size()}.
time.tex:The value held is unspecified if \tcode{d} is not in the range \crange{0}{255}.
time.tex:The value held is unspecified if \tcode{m} is not in the range \crange{0}{255}.
time.tex:always produces a remainder in the range of \crange{0}{11}.
time.tex:this operation results in a \tcode{month} holding a value in the range \crange{1}{12} even if \tcode{!x.ok()}.
time.tex:in the range \crange{months\{0\}}{months\{11\}}
time.tex:It can represent values in the range \crange{min()}{max()}.
time.tex:The value held is unspecified if \tcode{y} is not in the range \crange{-32767}{32767}.
time.tex:The value held is unspecified if \tcode{wd} is not in the range \crange{0}{255}.
time.tex:always produces a remainder in the range of \crange{0}{6}.
time.tex:this operation results in a \tcode{weekday} holding a value in the range \crange{0}{6} even if \tcode{!x.ok()}.
time.tex:in the range \crange{days\{0\}}{days\{6\}}
time.tex:The values held are unspecified if \tcode{!wd.ok()} or \tcode{index} is not in the range \crange{0}{7}.
time.tex:A \tcode{sys_days} in the range \crange{days\{-12687428\}}{days\{11248737\}}
time.tex:and \tcode{d_} is in the range \crange{1d}{(y_/m_/last).day()},
time.tex:If \tcode{ymd.day()} is in the range \crange{1d}{28d},
time.tex:and \tcode{ymd.day()} is not in the range \crange{1d}{28d},
time.tex:of the smallest possible integer in the range \crange{0}{18} such that
time.tex:The 12-hour equivalent of \tcode{h} in the range \crange{1h}{12h}.
time.tex:If \tcode{h} is not in the range \crange{0h}{23h},
time.tex:in the range \crange{0h}{11h},
time.tex:in the range \crange{12h}{23h},
time.tex:If \tcode{h} is not in the range \crange{1h}{12h},
time.tex:values in the range \crange{69}{99}
time.tex:and values in the range \crange{00}{68}
```

Reply:
I was wondering the same. I thought our range macros were for all kinds of ranges, not just iterators. By contrast, there's nothing "code" about `[a, b]`, and for a half-open range `[a, b)` would look outright weird, supposing that you associate "code font" with "balanced parentheses".

We can leave this as an open issue though, if this is controversial.

Reply:
See also: https://github.com/cplusplus/draft/pull/5647#discussion_r937433567.

Reply:
Thanks!

Reply:
Sorry, I misread, there's no code font involved here -- rather, it's whether we should manually spell out ranges in maths, or use a macro. That distinction seems to be largely unobservable in the output, so we can think about that later and consider the arguments for and against.